frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -180,13 +186,31 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 		return nil, err
 	}
 
-	for _, w := range dockerfile.Warnings {
-		opt.Warn(w.Short, w.URL, w.Detail, w.Location)
+	// Moby still uses the `dockerfile.PrintWarnings` method to print warnings,
+	// which prevents us from meaningfully refactoring the `parser.Parse` method
+	// to accept a lint warning callback in the same way that the `instructions.Parse`
+	// method does. As a result, we need to manually iterate over the warnings
+	// and call the lint warning callback for each one.
+	for _, warning := range dockerfile.Warnings {
+		// The `dockerfile.Warnings` *should* only contain warnings about empty continuation
+		// lines, but we'll check the warning message to be sure.
+		if warning.URL == linter.RuleNoEmptyContinuations.URL {
+			location := []parser.Range{*warning.Location}
+			msg := linter.RuleNoEmptyContinuations.Format()
+			linter.RuleNoEmptyContinuations.Run(opt.Warn, location, msg)
```

Comment:
nit. Maybe it would be more readable if organized to:
```
opt.Warn(linter.RuleNoEmptyContinuations, location, msg)
```
or 
```
linter := ... opt.Warn
linter.Show(lintrules.NoEmptyContinuations, location, msg)
```

Reply:
My pitch is that we let this be as is for this PR, and then let this + updating `moby`/`PrintWarnings` be a follow up PR.

Reply:
@tonistiigi Organizing the code this way basically does not square with how we're currently tying the rule message templates to the rule through generics—Go doesn't support parametrized methods, so neither

`type ConvertOpts struct {
....
    Warn: func[F any](LinterRule[F], ...)
}`

nor
`type Linter struct {
....
}
func (l *Linter)Show[F any](LinterRule[F}, ...) { ... }
`
are going to work because, in either case, the fix is to parameterize the containing struct with our `[F any]`, which is absolutely horribly terribly awfully nonfunctional in the case of `ConvertOpts` and merely ugly and ungainly in the case of a new `Linter` struct.